test/client_test.dart

```diff
@@ -75,6 +76,22 @@ void main() {
       expect(data?.user.userMetadata, {"Hello": "World"});
     });
 
+    test('Parsing invalid URL should emit Exception on onAuthStateChange',
+        () async {
+      expect(client.onAuthStateChange, emitsError(isA<AuthException>()));
+
+      const expiresIn = 12345;
+      const refreshToken = 'my_refresh_token';
+      const tokenType = 'my_token_type';
+      const providerToken = 'my_provider_token_with_fragment';
+
+      final urlWithoutAccessToken = Uri.parse(
+          'http://my-callback-url.com/welcome#expires_in=$expiresIn&refresh_token=$refreshToken&token_type=$tokenType&provider_token=$providerToken');
+      try {
+        await client.getSessionFromUrl(urlWithoutAccessToken);
+      } catch (_) {}
```

Comment:
`getSessionFromUrl` should throw, right? Let's test that as well, by adding a `fail()` after the method call.